pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -177,7 +178,12 @@ func (a *genericActuator) Reconcile(ctx context.Context, worker *extensionsv1alp
 		}
 	}
 
-	if err := a.updateWorkerStatus(ctx, worker, wantedMachineDeployments); err != nil {
+	machineImages, err := workerDelegate.GetMachineImages(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get the machine images")
+	}
+
+	if err := a.updateWorkerStatus(ctx, worker, wantedMachineDeployments, machineImages); err != nil {
```

Comment:
The reconciliation of a `worker` resource can take quite a while (especially if MCM has a sporadic downtime, etc.) which is why I think it's better to write the images rather at the beginning.

What's your opinion?

Reply:
Agreed, makes sense, will do!